exonum-java-binding-qa-service/src/main/java/com/exonum/binding/qaservice/QaServiceImpl.java

```diff
@@ -240,12 +248,12 @@ public StoredConfiguration getActualConfiguration() {
   }
 
   @SuppressWarnings("ConstantConditions") // Node is not null.
-  private HashCode submitTransaction(Transaction tx) {
+  private HashCode submitTransaction(RawTransaction rawTransaction) {
     checkBlockchainInitialized();
     try {
-      node.submitTransaction(tx);
-      return tx.hash();
-    } catch (InvalidTransactionException | InternalServerError e) {
+      node.submitTransaction(rawTransaction);
+      return rawTransaction.hash();
```

Comment:
Won't work. A client can't use it simply (= in queries for transaction messages), because it is a hash of payload, not of a whole transaction message.

@vitvakatu, @bullet-tooth  Shan't `node.submitTransaction` return a hash of the message that _core_ created? I'd expect it to.